### ISSUE TITLE
recording: make it possible to copy image descriptions when inserting from the gallery?

### ISSUE THREAD

**Issue by luizaogs** [arch-kiosk/arch-kiosk-office#2424]
This might be very specific to this project, but we keep having to insert the same image over and over again from the gallery with the same description every time, and having the option to insert it with its description would be very helpful. 

The reason why it might be specific to us is how we are recording walls, though if you guys want to think about expanding architectural recording it might be relevant to others, too. We take a picture of a wall and use the iPad photo editing tools to color the blocks so they are distinguishable, and then we give them numbers. Those numbers then correspond to created CM. So we then add that image to every CM record so that we have it as a reference when looking at information pertaining to specific blocks. Copying the CM record with the hamburger menu button does not solve the problem -- I tried and it copies the description but not the image.

Here's an example of one of the images in case the above description does not make sense: 

![IMG_0417](https://github.com/arch-kiosk/arch-kiosk-office/assets/67970402/aaa86e59-ef36-481d-a038-c6100d571651)

**Comment by urapadmin** [comment]
I have an idea. I can show all the description that have been used for an image (they are context-bound) in the gallery next to the identifier. When using a green apply button of a specific description that text will be used when the image gets applied to the context. 

That should do what you want, no?

**Comment by urapadmin** [comment]
would look like this:
![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/38838314/13180533-7240-4b72-b8e7-b5bcb42e5d88)

**Comment by luizaogs** [comment]
Yup, that looks good to me.

**Comment by urapadmin** [comment]
comes too late, though, doesn't it?

**Comment by luizaogs** [comment]
For this season yes :)

**Comment by urapadmin** [comment]
### recording 15.11
needs to be tested in any of these image containers:
- [ ] dayplans, 
- [ ] locus_photo, 
- [ ] collected_material_photo
- [ ] pottery_images
- [ ] su_ceramics_images
- [ ] coordinate photo

For the others (e.g. locus relations, site note images) selecting a specific text turns out the same as selecting without a text

**Comment by luizaogs** [comment]
Not entirely working in collected material photos yet; some get the description and some do not (the first image I added from the gallery in that layout did not, the second did, the third did not).

**Comment by luizaogs** [comment]
Dumb question - are pottery images the warecode analysis images?

**Comment by urapadmin** [comment]
yup

**Comment by urapadmin** [comment]
> Not entirely working in collected material photos yet; some get the description and some do not (the first image I added from the gallery in that layout did not, the second did, the third did not).
> 
> Not entirely working in su ceramics, either: the first two images I added did not get their attached descriptions, but the third one I added did.
> 
> And something similar for coordinate photos, too: the first one I added did not get its description, the second did.

hm. Can't reproduce that on Windows (and there is no difference between loci, dayplans and collected material), so I wonder if it is an iPad or a timing issue and it would eventually appear with loci and dayplans as well?!? Did you add to a new empty container? Or was there something in there? We have to find a pattern here.

**Comment by urapadmin** [comment]
### recording 15.11.1
- I have made a tiny little change for collected materials. Please see, if you can still reproduce it there.

**Comment by luizaogs** [comment]
I added new empty containers in all cases but was testing on my Mac. Let me see if the iPad is the same -- I am still having issues in collected material and just tried pottery photos and had the same problem there, too.

**Comment by luizaogs** [comment]
Yeah weirdly they all work on the iPad but on my Mac the only ones working reliably, it seems, are locus and dayplans. Not sure why...

**Comment by urapadmin** [comment]
argl. There is our next image container nightmare. I can not see this at all on Windows so we have a FileMaker on Mac speciality here. What FM and MacOS Version are you on?

**Comment by luizaogs** [comment]
Mac OS 14.2.1, FM 19.6.3.

**Comment by urapadmin** [comment]
do you get the feeling it it makes any difference if you do it all quite slowly (waiting a bit between two images)?

**Comment by luizaogs** [comment]
Argh now I tried first in dayplans and that description didn't work either, so I take back what I said about dayplans and loci being more reliable.

**Comment by luizaogs** [comment]
Ok wait maybe I found a pattern - the description does not seem to copy if it has been inserted in a previous image container. But I am not currently having issues inserting images with descriptions towards the bottom of the list in the gallery, which suggests unused descriptions work.

**Comment by urapadmin** [comment]
> Ok wait maybe I found a pattern - the description does not seem to copy if it has been inserted in a previous image container. But I am not currently having issues inserting images with descriptions towards the bottom of the list in the gallery (into image containers in any of those forms), which suggests that unused descriptions work.

hm. I wonder if that holds. At least it does not make immediate sense to me. Differences between bottom / top of a list might sounds a bit more promising. Or e.g. do you see a difference between manipulating a container that already had an image and one that you just freshly created?

**Comment by luizaogs** [comment]
Oh, more interesting: inserting an image with description from the gallery into a container with a description that had an image results in the new gallery image being inserted but the description from the image that had occupied the container previously stays there, instead of the correct one.

**Comment by luizaogs** [comment]
And actually, it seems like I consistently have issues adding the first two images in the gallery with their descriptions (if sorted by description in PVD) but not ones below them...

**Comment by urapadmin** [comment]
I was wondering. Have you been testing this with PVD or BUAP?

**Comment by urapadmin** [comment]
sorry, MADP, I meant.

**Comment by luizaogs** [comment]
PVD. You told me not to test on live servers or databases! I listen to you :)

**Comment by urapadmin** [comment]
I abandoned this (despair I guess) but it is rolled out. I should get back to it.

**Comment by luizaogs** [comment]
With MAD-P now actually happening, do you think you might go back to this? I think I'll be using it a lot.

**Comment by urapadmin** [comment]
The functionality is actually implemented and then it turned into an unhappy ticket. I put it on the MAD-P list.

**Comment by urapadmin** [comment]
I was thinking about the original use-case. Wouldn't it make most sense for you to use (and see) the context-independent description? I am not sure how to achieve that given that originally you'd always input the image and a description in a context but from a data modelling perspective we are creating a lot of redundancy here. At least I suspect that normally you would have the same one description for such an image and then reduplicate it over and over again.

**Comment by luizaogs** [comment]
Hm I don't think so? The reason I need this is because recording walls means recording a bunch of blocks as CM, and for each CM I want the annotated photo in there. So the original description, something like "Annotated photo of wall A," works just fine. Unless I'm misunderstanding your question, which in my current state is very possible.

**Comment by urapadmin** [comment]
that is exactly what I mean. You don't need a different description for the image in the contexts the image gets inserted. If you had access to the context-independent description and see the context independent description in FileMaker with the image, your use case would be covered. 

It is just that we don't use the context-indepenent description in FileMaker. I am wondering how I could change that.

**Comment by luizaogs** [comment]
Oh I see. Yes, you're right.

**Comment by urapadmin** [comment]
- see #2984

**Comment by urapadmin** [comment]
I keep thinking about this and the pain it is for you in the field. Given that image annotation never happened and I don't even see myself getting to it, I wonder what we can do here.

For u2ers I experimented with something: There no context-dependent image description exists at all, the u2ers recording app stores right into the main image description - the context independent description.  

In MAD-P the feature we were talking about in this ticket is, if I understand it correctly, simply creating a ton of redundancies by copying the same context-dependent image description into other contexts over and over again. You don't vary the description of the same image by context, do you? 

So what I am thinking is whether I can change the regular recording app in a way that it can operate in the u2ers mode (if configured to do so, of course). The result would be that whenever you insert an image into a context, the description will be right there. If you change the description in one place, it will change automatically in all contexts where that image appears (that is exactly what we normally try to avoid). 
Would that make sense? Of course, it would work that way for ALL images, so you won't have the chance anymore to have an image in two different contexts with different descriptions.

I hope I described this intelligibly.

**Comment by luizaogs** [comment]
Hm I'm not sure. I am looking through the FR and we sometimes have different context dependent and independent descriptions for the same photo, and we used them to say different things (the independent description was something we added when going through images in the FR itself, of course, so we sometimes added additional details that we hadn't added in FM itself). So with this suggested solution, I guess we wouldn't have the dependent descriptions? Right? I worry that will create more problems that I can't predict right now. 

Ideally we would be able to have both types of descriptions in FM, though not always showing both - that sounds messy. Perhaps the independent one could be triggered for some images in FM and not others (since, you're right, we're creating redundancies by inserting the same description). And those would be the ones for the case as described in this ticket, where we're just reusing the same description in a bunch of places. But I assume that is too difficult/impossible to do.

**Comment by urapadmin** [comment]
I think that would be an example of both having ones cake and eat it. At least I can't imagine how to guide your input in the UI to have both worlds.

Sounds to me as if I'd better go try to fix the bugs and we live with the redundancies. Looking at the file repository for MADP is painful for the database designer in me. But without image annotation the recording you do on that project is just not what Kiosk is built for. So I am a bit out of ideas how to make life easier for you.

**Comment by luizaogs** [comment]
Looking at it is painful for me, too, but our data recording is more complex than u2ers so I don’t think having one type of description works. (We are excavating and surveying, so it is what Kiosk is built for, but the architectural stuff we need to do goes beyond what the system can support.)

It’s fine - I lived with it for a season, already, so will do so again.

**Comment by urapadmin** [comment]
I cannot reproduce this. Having added images and descriptions like a mad man, works totally reliable in all the places for me.
And I don't even have an idea right now what I could possibly tweak preemptively. 
So one thing you could try on your side: Update FileMaker to at least 20 or 21. Do you have access to a higher FM Version?

**Comment by urapadmin** [comment]
@lbestock: Could you please join the test here?

**Comment by urapadmin** [comment]
can't let it go.
I swear, this is my last experiment with it. I have made this change:
- in collected material photos I have added the context independent description to the list of images in the lower left corner. This is only a display
- if you open the popup for that image you find the independent description in the lower part - but here you can edit it. 
- in the gallery if you add a photo the first button is always the one that adds the photo without a context dependent text. That has always been the case but now you see the context INdepenent description right next to it. If you use that button not context dependent description will be added to the image (as has always been the case, just the independent description never showed).

The idea is that when you first make an image you add a context independent description, the one that would fit for most photos. 
If in a certain context you need more than that you add a context independent description. 

Currently this experiment is limited to collected material and it is hidden behind a config key: You have to set "experimental_mode" in "admin" to "yup".

Give it a try. If you think that is useful for MADP, it isn't too much work to add it to the other image containers and hide the feature behind a normal config key.

**Comment by urapadmin** [comment]
### recording app 16.15
- see above
- I have also change the subtlest thing in copying over the description. But that difference only existed for collected material photos and I don't see how it could possibly have the buggy effects that you were experiencing. However, I changed it.

**Comment by luizaogs** [comment]
Updated FM so will test this now and let you know. Thank you.

**Comment by luizaogs** [comment]
Hmm I don't think the change came through to the 16.15 version? At least I can't find the config key or see the independent description in CM.

**Comment by urapadmin** [comment]
the config key has always been there, I was wrong initially. It is not in the general section but in the admin section, But it really should be there.

**Comment by luizaogs** [comment]
Do you mean that we have another configuration that is not the regular one? Do I have amnesia? Good lord.

**Comment by luizaogs** [comment]
This is in advanced config

<img width="652" height="498" alt="Image" src="https://github.com/user-attachments/assets/7b8914f5-460e-4099-b664-ae798f8b3fd4" />

**Comment by urapadmin** [comment]
don't ask me why but the "admin" section is not being searched

**Comment by urapadmin** [comment]
<img width="1076" height="652" alt="Image" src="https://github.com/user-attachments/assets/b4ef59d1-6754-4340-b9f7-3dabc1eb3138" />

**Comment by luizaogs** [comment]
Ohh. I'm dumb. Ok.

**Comment by urapadmin** [comment]
and I already found the first bug here: The context independent description must not show if there is not image.

**Comment by luizaogs** [comment]
> The idea is that when you first make an image you add a context independent description, the one that would fit for most photos.
> If in a certain context you need more than that you add a context independent description.

Just to make sure I understand: you mean, if in a certain context I need more than that I would then add a DEPENDENT description. Right? So the idea being that for these damn wall photos I only need the independent, but then if I happen to want something extra in a specific context I have the option to add that also.

**Comment by luizaogs** [comment]
Anyway, yes, this looks really good (besides the bug you mention up there). 

I can change the "a description would be nice" text, right? I'm thinking it might be good to have that say "context dependent description" instead.

**Comment by urapadmin** [comment]
you might want to look at the file repository and query and view, though, and think about whether the way it shows the independent description and combines independent with a dependent description is suitable. I am not entirely sure that both would be shown properly. That can be configured, I think. But it is an ooooold option from 2018 that I have never used :o)

Otherwise if you think this will make your life easier and allow us to avoid a few redundancies, I would apply it at least to dayplans and loci. Those are the places you use, right? Wasn't there also something about site notes (but those don't have an independent description to my knowledge).

**Comment by urapadmin** [comment]
@lbestock I think it is better to hide this feature with a config key by default. My assumption always is that the mere difference between context-dependent and independent description is too grasp for most Kiosk users. Agreed? Or opening it up for everybody. This is the first time that the independent description becomes readily available within the recording app.

**Comment by luizaogs** [comment]
Q&V in PVD isn't working (time zone issue, I think I saw a ticket about it) but in the FR they work well together. So yes, I think this would be great! Yes, CM, dayplans and loci are the three I can think of that would benefit from this (CM being the most important).

**Comment by luizaogs** [comment]
The site notes issue of relevance for MADP is this one: #2377 

But not related to this, I don't think.

**Comment by lbestock** [comment]
> [@lbestock](https://github.com/lbestock) I think it is better to hide this feature with a config key by default. My assumption always is that the mere difference between context-dependent and independent description is too grasp for most Kiosk users. Agreed? Or opening it up for everybody. This is the first time that the independent description becomes readily available within the recording app.

Absolutely hide this with a config key by default. I think this is rocket science level stuff.

**Comment by urapadmin** [comment]
💩

**Comment by urapadmin** [comment]
🚀  💩

**Comment by luizaogs** [comment]
🫣🤡💀

**Comment by urapadmin** [comment]
- this is now also solving #2984

**Comment by luizaogs** [comment]
Did you release an update for this?

**Comment by urapadmin** [comment]
no, still on it

**Comment by luizaogs** [comment]
Ok. I asked because it was listed in the release  notes.

**Comment by urapadmin** [comment]
since we are doing this I move it to a new ticket as the original ticket is about something else: #3430

**Comment by urapadmin** [comment]
no no no, no closing of this ticket here. Not until we either get rid of the functionality entirely or make it work reliably.

**Comment by luizaogs** [comment]
Ha oh right, I was trying to make you less stabby. Woops.

**Comment by urapadmin** [comment]
doesn't work that way with me. But you may check if the issue is still the case for you. I still cannot reproduce this and @lbestock is drowning in emails.

**Comment by luizaogs** [comment]
Perhaps I am having a stroke? 

If the new independent description key is set to nope, I am able to insert any image description on the iPad. But I can't do that on my Mac?

Both with "nope," this is what my Mac looks like: 

<img width="749" height="429" alt="Image" src="https://github.com/user-attachments/assets/2a07bccb-0138-477b-b660-5642aeaeeaa5" />

My iPad: 

<img width="2160" height="1620" alt="Image" src="https://github.com/user-attachments/assets/ded1cef2-d78f-4b0f-ac49-d3ae417adacc" />

**Comment by urapadmin** [comment]
the setting does not change the behaviour in the gallery. You can insert the context dependent description no matter if there is a nope or a yup in the setting.

You should, that is ...
The only way I can make it appear like that is when I go to the gallery straight on the start page.

**Comment by urapadmin** [comment]
but then the other green button would be gone, too.

So no, looks like you have no descriptions.

Can you please check if you can search for one of those loci that don't have a green button by something in their description?

**Comment by luizaogs** [comment]
Hm these are workstations maybe downloaded at different times, now that I think about it. Which may account for the descriptions on one device and not on the other.

I’ll reprepare and redownload both and check.

**Comment by luizaogs** [comment]
Something is wrong with the descriptions in the gallery. See this photo, for instance: c6ab56ab-354b-4f79-8850-9de1f6d544cb 

In the file repository it has a description, as it should. 

In the gallery .... 

<img width="701" height="219" alt="Image" src="https://github.com/user-attachments/assets/45a1e395-f9ab-4d86-b7e3-f99e840e9def" />

**Comment by urapadmin** [comment]
I suspect this is a test server issue. But I'll look into it. Does none of the images have descriptions?

**Comment by luizaogs** [comment]
No, I don't think so.

**Comment by urapadmin** [comment]
there are virtually no image texts in the file identifier cache

**Comment by urapadmin** [comment]
refreshing the file identifier cache has not helped

**Comment by urapadmin** [comment]
we never get to test this properly because of other bugs getting in the way. The last one: #3435 
When that new Kiosk version is out you can find what's next.

**Comment by luizaogs** [comment]
Hallelujah. 

Yes, I have made sure that I can no longer reproduce the issues above.
